#167336013 feature(PostProperty): implement creat property by agent

swaibat opened the pull request:
#### What does this PR do?
allows only the agent to creates a new property in the database

#### Description of Task to be completed?
-  implement Create property feature
-  add middleware to stop users from creating a property

#### How should this be manually tested?.
- Clone the repo
- run `npm i ` to install the packages 
- run `npm run test` to test the post property route

#### What are the relevant pivotal tracker stories?
[#167336013](https://www.pivotaltracker.com/story/show/167336013)
#### Screenshots (if appropriate)
![only Agent](https://user-images.githubusercontent.com/41104288/61361415-c374c200-a880-11e9-9708-10c0920c102e.PNG)
![create property](https://user-images.githubusercontent.com/41104288/61361434-c8d20c80-a880-11e9-9e7a-57fa4b8724ea.PNG)

[Finishes #167336013]

swaibat committed:
feature(PostProperty): implement creat property by

-  implement Create property feature
-  add a middleware to stop users from creating a property

[Finishes #167336013]

swaibat committed:
ch(UpdateHelpers):create and Add errors helpers

-  create errors helpers function

[Finishes #167336013]

Oluwasegun-AA commented on apiV2/controller/propertyController.js:
```diff
@@ -1,41 +1,47 @@
+    const property = adObj.addProperty(adObj);
+    property.then((e) => {
+      console.log(req.path);
+      return Resp(201, 'Property created', e.rows[0], res);
```
please give this a more descriptive name.. resp is not descriptive.. also you do not start a function name with an upper case letter (airbnb guide). please make adjustment

Oluwasegun-AA commented on apiV2/controller/propertyController.js:
```diff
@@ -1,41 +1,47 @@
+    property.then((e) => {
+      console.log(req.path);
+      return Resp(201, 'Property created', e.rows[0], res);
+      // res.status(201).send({ status: 201, property: e.rows[0] });
```
@swaibat , please remove the redundant set of codes in your workflow

Oluwasegun-AA commented on apiV2/services/db.js:
```diff
@@ -1,14 +1,13 @@
+// ternary operator
+const client = process.env.NODE_ENV === 'test'
+  ? new Client({ connectionString: process.env.TESTDB_URL })
+  : new Client({ connectionString: process.env.DATABASE_URL });
```
@swaibat , good Job implementing this

Oluwasegun-AA commented:
please address the feedback and merge afterwards

swaibat committed:
bug(fix-errors): imolement cammel case and remove redadant code

-  assigned descriptive names
-  fixed camnelCase propblems

[Finishes #167336013]
